packages/shared/src/hooks/squads/useSquadPrivacyOptions.tsx

```diff
@@ -0,0 +1,108 @@
+import { Anchor } from '../../components/text';
+
+interface UseSquadPrivacyOptionsProps {
+  totalPosts: number;
+  status: SquadStatus;
+}
+
+const badge: Record<SquadStatus, ReactNode> = {
+  [SquadStatus.InProgress]: <LockIcon size={IconSize.Small} />,
+  [SquadStatus.Pending]: <TimerIcon size={IconSize.Small} />,
+  [SquadStatus.Rejected]: <BlockIcon size={IconSize.Small} />,
+  [SquadStatus.Approved]: null,
+};
+
+const classes = {
+  wrapper: 'border border-border-subtlest-tertiary rounded-16 p-4 pt-1 w-full',
+  content: 'w-fit',
+};
+
+const PUBLIC_SQUAD_REQUIRED_POSTS = 3;
+
+export enum PrivacyOption {
+  Private = 'private',
+  Public = 'public',
+}
+
+export const useSquadPrivacyOptions = ({
+  totalPosts,
+  status,
+}: UseSquadPrivacyOptionsProps): RadioItemProps<PrivacyOption>[] =>
+  useMemo(() => {
+    const isInProgress = status === SquadStatus.InProgress;
+    const isPending = status === SquadStatus.Pending;
+
+    if (isNullOrUndefined(totalPosts)) {
+      return [];
+    }
+
+    return [
+      {
+        label: 'Private',
+        value: PrivacyOption.Private,
+        className: classes,
+        afterElement: (
+          <StatusDescription className="ml-9">
+            Only people who join the squad can see the content.
+          </StatusDescription>
+        ),
+      },
+      {
+        label: 'Public',
+        value: PrivacyOption.Public,
+        disabled: status !== SquadStatus.Approved,
+        className: classes,
+        afterElement: (
+          <div
+            className={classNames(
+              'relative grid w-full grid-cols-1 gap-5',
+              isInProgress && 'tablet:grid-cols-2',
+            )}
+          >
+            <span
+              className={classNames(
+                'absolute -top-1 right-0 flex -translate-y-full flex-row items-center',
+                isPending ? 'text-text-tertiary' : 'text-text-disabled',
+              )}
+            >
+              {badge[status]}
+              {isPending && (
+                <span className="ml-1 font-bold typo-callout">Pending</span>
+              )}
+            </span>
+            <StatusDescription className="ml-9 mt-2">
+              Everyone can see the content, and the posts may appear on the main
+              feed.
+              <Anchor
+                href={squadsPublicGuide}
+                rel="noopener noreferrer"
+                target="_blank"
+                className="mt-1 block"
+              >
+                Read more about Public Squads
+              </Anchor>
+            </StatusDescription>
+            {/* it seems to be a bit unusual to show 50/3, hence we max the count at 3 so the progress will always be 3/3 */}
+            {isInProgress && (
+              <PublicStatusPanel
+                count={Math.min(totalPosts, PUBLIC_SQUAD_REQUIRED_POSTS)}
+                required={PUBLIC_SQUAD_REQUIRED_POSTS}
+              />
+            )}
+          </div>
```

Comment:
This can be separated in a component

Reply:
I mean this was the purpose of putting it in a hook, to move it somewhere else, but yeah, we can do that too.

Reply:
Yeah, still makes the code messy a bit, maybe it would help to have it separated in a component even if in the same file

Reply:
Yep. Much cleaner indeed! 🚀